### PR TITLE
[Java] refine Collection util data structure (#1287)

### PR DESCRIPTION
**Pull request by mof-dev-3**
1. use local reference when possible
2. make tuple value field final

### CODE REVIEW

File: java/fury-benchmark/src/main/java/org/apache/fury/benchmark/Identity2IdMap.java [hunk 1]
```diff
@@ -26,16 +26,16 @@
 public class Identity2IdMap {
   private static final int RESERVE = 4;
   private static final int MAX_DEPTH = 4;
-  static int[] prim = {
```
Comment (chaokunyang) [inline java/fury-benchmark/src/main/java/org/apache/fury/benchmark/Identity2IdMap.java:29]: This class is copied from fst, it's unnecessary to update it

File: java/fury-core/src/main/java/org/apache/fury/collection/FuryObjectMap.java [hunk 2]
```diff
@@ -180,6 +180,7 @@ public V put(K key, V value) {
   /** Skips checks for existing keys, doesn't increment size. */
   private void putResize(K key, V value) {
     K[] keyTable = this.keyTable;
+    V[] valueTable = this.valueTable;
```
Comment (chaokunyang) [inline java/fury-core/src/main/java/org/apache/fury/collection/FuryObjectMap.java:183]: This is accessed only once, it's unnecessary to make it as a local variable

File: java/fury-core/src/main/java/org/apache/fury/collection/ObjectIntMap.java [hunk 3]
```diff
@@ -92,6 +92,7 @@ public void put(K key, int value) {
 
   private void putResize(K key, int value) {
     K[] keyTable = this.keyTable;
+    int[] valueTable = this.valueTable;
```
Comment (chaokunyang) [inline java/fury-core/src/main/java/org/apache/fury/collection/ObjectIntMap.java:95]: Accessed only once, unnecessary to use a local variable

File: java/fury-core/src/main/java/org/apache/fury/collection/Tuple2.java [hunk 4]
```diff
@@ -46,6 +48,43 @@ public Tuple2(T0 value0, T1 value1) {
     this.f1 = value1;
   }
 
+  public static class NonFinal<T0, T1> implements Serializable {
```
Comment (chaokunyang) [inline java/fury-core/src/main/java/org/apache/fury/collection/Tuple2.java:51]: Make this as a top level class, maybe name it as MutableTuple2?

File: java/fury-core/src/main/java/org/apache/fury/collection/Tuple3.java [hunk 5]
```diff
@@ -50,6 +52,70 @@ public Tuple3(T0 value0, T1 value1, T2 value2) {
...
+   * nonFinal.
+   *
+   * @param value0 The value for field 0
+   * @param value1 The value for field 1
+   * @param value2 The value for field 2
+   */
+  public static <T0, T1, T2> NonFinal ofNonFinal(T0 value0, T1 value1, T2 value2) {
+    return new NonFinal<>(value0, value1, value2);
+  }
+
```
Comment (chaokunyang) [inline java/fury-core/src/main/java/org/apache/fury/collection/Tuple3.java:66]: Ditto

File: java/fury-benchmark/src/main/java/org/apache/fury/benchmark/Identity2IdMap.java (hunk 1 above)
Comment (mof-dev-3) [reply java/fury-benchmark/src/main/java/org/apache/fury/benchmark/Identity2IdMap.java]: Done. revert all change in this class

**Review by chaokunyang — APPROVED** [review]
LGTM

**Pull request merged**